docs/trace/exception-handling/Program.cs

```diff
@@ -0,0 +1,77 @@
+// <copyright file="Program.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Diagnostics;
+using OpenTelemetry;
+using OpenTelemetry.Trace;
+
+public class Program
+{
+    private static readonly ActivitySource MyActivitySource = new ActivitySource(
+        "MyCompany.MyProduct.MyLibrary");
+
+    public static void Main()
+    {
+        AppDomain.CurrentDomain.UnhandledException += UnhandledExceptionHandler;
+
+        using var tracerProvider = Sdk.CreateTracerProviderBuilder(options =>
+            {
+                options.SetErrorStatusOnException = true;
+            })
+            .SetSampler(new AlwaysOnSampler())
+            .AddSource("MyCompany.MyProduct.MyLibrary")
+            .AddConsoleExporter()
+            .Build();
+
+        try
+        {
+            Func();
+        }
+        catch (Exception)
+        {
+            // swallow the exception
+        }
+
+        Func();
+    }
+
+    private static void Func()
+    {
+        using (MyActivitySource.StartActivity("Foo"))
+        {
+            using (MyActivitySource.StartActivity("Bar"))
+            {
+                throw new Exception("Oops!");
+            }
+        }
+    }
+
+    private static void UnhandledExceptionHandler(object source, UnhandledExceptionEventArgs args)
```

Comment:
If user sets `SetErrorStatusOnException=true` should we register this handler automatically?

Reply:
My answer would be no:
1. The side effect might be unacceptable for some mission critical applications (e.g. library should not be too invasive).
2. `UnhandledExceptionHandler` has a bigger scope - it will touch everything (e.g. including activities that are not subscribed by the TracerProvider).